### ISSUE TITLE
Attribution control does not collapse on small devices

### ISSUE THREAD

**Issue by gzurbach** [maplibre/maplibre-gl-js#1799]
**maplibre-gl-js version**: 2.4.0

**browser**: Chrome 106

### Steps to Trigger Behavior

 1. Create map without attribution control:

```js
const map = new maplibregl.Map({ 
  ...
  attributionControl: false,
});
```

 2. Add an attribution control manually, forcing compact behavior:

```js
map.addControl(new maplibregl.AttributionControl({ compact: true }));
```

### Expected Behavior

On devices smaller than 640px wide, the attribution control should be collapsed by default. 
At the very least, passing `compact: true` should force this behavior.

### Actual Behavior

The attribution control is not collapsed. Not by default, not even by passing `compact: true`:

<img width="365" alt="Screen Shot 2022-10-31 at 4 53 58 PM" src="https://user-images.githubusercontent.com/795927/199053251-fe0d4878-2dd2-446c-b0e6-59058ddc827d.png">

Note: in this example the device is 390px wide.

The attribution control *does* collapse when clicking on the control itself. But there is no way to force the initial state of the control.

**Edit**: I am pretty sure the bug has something to do with this method: https://github.com/maplibre/maplibre-gl-js/blob/a3568e0363c7fd3540a1e18163e70665efbd084f/src/ui/control/attribution_control.ts#L174-L188 ... I will try to take a look this week if I have some time.

**Comment by acalcutt** [comment]
This was intended behavior after https://github.com/maplibre/maplibre-gl-js/pull/795

when loading a map, the compact button shows open to comply with the spirit of openstreetmap attribution. there was discussion that having attibution default not vistible. did not comply.

As soon as you start moving on the map, the button should close, and you would just see the button.

**Comment by gzurbach** [comment]
Ah that makes a lot of sense! I didn't even realize it collapsed when I moved the map. I guess the [documentation](https://maplibre.org/maplibre-gl-js-docs/api/markers/#attributioncontrol) could make this a bit more clear?

Replace:

> An AttributionControl control presents the map's attribution information.

By:

> An AttributionControl control presents the map's attribution information. By default the attribution control is expanded (regardless of device width). On smaller devices, the attribution control will collapse when the user moves the map.

And replace:

> If true , force a compact attribution that shows the full attribution on mouse hover.

By:

> If true, the attribution control will collapse on all devices when moving the map.

I can submit a PR if that seems appropriate, let me know.

**Comment by HarelM** [comment]
Sounds good, thanks!

**Comment by HarelM** [comment]
Closed via #1801

**Comment by FutureBuddha** [comment]
is there a way to make it closed as default?
